plugins/transport-grpc/src/main/java/org/opensearch/transport/grpc/GrpcPlugin.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.transport.grpc;
+
+import org.opensearch.common.network.NetworkService;
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.core.indices.breaker.CircuitBreakerService;
+import org.opensearch.plugins.NetworkPlugin;
+import org.opensearch.plugins.Plugin;
+import org.opensearch.telemetry.tracing.Tracer;
+import org.opensearch.threadpool.ThreadPool;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import static org.opensearch.transport.grpc.Netty4GrpcServerTransport.GRPC_TRANSPORT_SETTING_KEY;
+import static org.opensearch.transport.grpc.Netty4GrpcServerTransport.SETTING_GRPC_BIND_HOST;
+import static org.opensearch.transport.grpc.Netty4GrpcServerTransport.SETTING_GRPC_HOST;
+import static org.opensearch.transport.grpc.Netty4GrpcServerTransport.SETTING_GRPC_PORTS;
+import static org.opensearch.transport.grpc.Netty4GrpcServerTransport.SETTING_GRPC_PUBLISH_HOST;
+import static org.opensearch.transport.grpc.Netty4GrpcServerTransport.SETTING_GRPC_PUBLISH_PORT;
+import static org.opensearch.transport.grpc.Netty4GrpcServerTransport.SETTING_GRPC_WORKER_COUNT;
+
+/**
+ * Main class for the gRPC plugin.
+ */
+public final class GrpcPlugin extends Plugin implements NetworkPlugin {
+
+    /**
+     * Creates a new GrpcPlugin instance.
+     */
+    public GrpcPlugin() {}
+
+    @Override
+    public Map<String, Supplier<AuxTransport>> getAuxTransports(
+        Settings settings,
+        ThreadPool threadPool,
+        CircuitBreakerService circuitBreakerService,
+        NetworkService networkService,
+        ClusterSettings clusterSettings,
+        Tracer tracer
+    ) {
+        return Collections.singletonMap(
+            GRPC_TRANSPORT_SETTING_KEY,
+            () -> new Netty4GrpcServerTransport(settings, Collections.emptyList(), networkService)
```

Comment:
Apologies if it was discussed before, why do we need `Supplier<AuxTransport>` vs `AuxTransport`? Why this level of indirection is needed?

Reply:
Since `getAuxTransports` is called for every plugin implementing an auxiliary transport it makes sense to return a supplier here as we may end up leaving any number of these transports [un-used based on settings](https://github.com/finnegancarroll/OpenSearch/blob/grpc/server/src/main/java/org/opensearch/common/network/NetworkModule.java#L380). This avoids unnecessary work or executing constructor code of an aux transport which may be discarded.

Reply:
Sure, makes sense, thank you